src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String username, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(username);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Incorrect username or password, try again");
```

Comment:
```suggestion
            throw new AuthenticationException("Incorrect username, try again");
```
Here you can definitely say it, because the user was not found in DB by provided username.